runtime/str.go

```diff
@@ -264,6 +294,17 @@ func strHash(f *Frame, o *Object) (*Object, *BaseException) {
 	return h.ToObject(), nil
 }
 
+func strIndex(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	index, raised := strFind(f, args, nil)
```

Comment:
If checkMethodArgs() raises an exception then it will say `'find' of 'str' requires ...` rather than `'index' of 'str' ...`

CPython gets around this by giving a really generic error message: `TypeError: find/rfind/index/rindex() takes at least 1 argument (0 given)`